app/src/main/java/io/github/sds100/keymapper/constraints/ConstraintState.kt

```diff
@@ -11,3 +12,41 @@ data class ConstraintState(
     val constraints: Set<Constraint> = emptySet(),
     val mode: ConstraintMode = ConstraintMode.AND,
 )
+
+// Comparator.reversed() requires API level 24
+class KeyMapConstraintsComparator(
+    private val reverse: Boolean = false,
+) : Comparator<KeyMap> {
+    override fun compare(
+        keyMap: KeyMap?,
+        otherKeyMap: KeyMap?,
+    ): Int {
+        // TODO
+
+        if (keyMap == null || otherKeyMap == null) {
+            return 0
+        }
+
+        for (i in 0 until keyMap.constraintState.constraints.size.coerceAtMost(otherKeyMap.constraintState.constraints.size)) {
```

Comment:
Can you put this into a separate variable and maybe explain what coerceAtMost is doing?